src/sentry_session.c

```diff
@@ -219,6 +221,9 @@ sentry__add_current_session_to_envelope(sentry_envelope_t *envelope)
     SENTRY_WITH_SCOPE_MUT (scope) {
         if (scope->session) {
             sentry__envelope_add_session(envelope, scope->session);
+            // we're assuming that if a session is added to an envelope it
+            // will be sent onwards.  This means we now need to set the init
+            // flag to false because we're no longer the initial session update.
```

Comment:
I’m still wondering if this is better done on the server, based on the uuid and timestamps?

Reply:
No. That's an important optimization for the data model that cannot be done on the server.